src/player.js

```diff
@@ -636,6 +636,33 @@ export default class Player extends FakeEventTarget {
     this.dispatchEvent(new FakeEvent(CustomEventType.PLAYER_DESTROY));
     this._eventManager.destroy();
   }
+  /**
+   * Listen to the video element events and triggers them from the engine.
+   * @private
+   * @param {FakeEvent} event data
+   * @param {?boolean} stopSeeking don't seek to the last detach point
+   * @returns {void}
+   */
+  _attachMediaSource(event: FakeEvent, stopSeeking: ?boolean): void {
+    if (this._engine) {
+      this._engine.attachMediaSource(stopSeeking);
+    }
+  }
+
+  /**
+   * Listen to the video element events and triggers them from the engine.
+   * @private
+   * @returns {void}
+   */
+  _detachMediaSource(): void {
+    if (this._engine) {
+      if (this._engine.ended) {
+        this._eventManager.unlisten(this, AdEventType.AD_BREAK_END, this._attachMediaSource);
+        this._eventManager.listen(this, AdEventType.ADS_COMPLETED, e => this._attachMediaSource(e, true));
```

Comment:
```suggestion
        this._eventManager.listen(this, AdEventType.ALL_ADS_COMPLETED, e => this._attachMediaSource(e, true));
```
`ADS_COMPLETED` is per ads plugin